src/DIRAC/WorkloadManagementSystem/DB/ElasticJobParametersDB.py

```diff
@@ -134,7 +134,7 @@ def setJobParameter(self, jobID, key, value):
 
         self.log.debug("Inserting data in %s:%s" % (self.indexName, data))
 
-        result = self.index(self.indexName, body=data, docID=str(jobID) + key)
+        result = self.index(self.indexName, body=data, docID=(str(jobID) + key)[:511])
```

Comment:
Can you add a comment explaining why this is the case? Are there any potential issues with data getting collapsed into a single record?

Reply:
I also wonder what job parameter key can exceed 512 characters. You certainly have an example. If we hit this limit, we have a problem anyway, I think

Reply:
The `_id` in ES can't exceed 512 bytes, this is a ES hard-coded limitation. The potential issue is an extremely rare case of having 2 identical `_id`s. If we are afraid of that, we can of course hash, but we would lose the ability to retrieve in the simplest manner the documents for "obvious" cases. What is your opinion?

Reply:
I don't know what these IDs look like to be able to understand what the failure mode would be like and give an good opinion.

> we would lose the ability to retrieve in the simplest manner the documents for "obvious" cases

Could we keep the simplicity using something like this:

```python
docID = str(jobID) + key
if len(docID) > 505:
    docID = docID[:475] + hashlib.md5(docID.encode()).hexdigest()[:35]
```

Programmatically retrieving keys won't work but at least no information will be lost.